src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/ScanBatch.java

```diff
@@ -94,29 +93,29 @@ public ScanBatch(final File signatureScannerInstallDirectory, final File outputD
     /**
      * The default install directory will be used if the batch does not already have an install directory.
      */
-    public List<ScanCommand> createScanCommands(final File defaultInstallDirectory, final ScanPathsUtility scanPathsUtility, final IntEnvironmentVariables intEnvironmentVariables) throws BlackDuckIntegrationException {
+    public List<ScanCommand> createScanCommands(File defaultInstallDirectory, ScanPathsUtility scanPathsUtility, IntEnvironmentVariables intEnvironmentVariables) throws BlackDuckIntegrationException {
         String scanCliOptsToUse = scanCliOpts;
         if (null != intEnvironmentVariables && StringUtils.isBlank(scanCliOptsToUse)) {
-            final String scanCliOptsEnvironment = intEnvironmentVariables.getValue("SCAN_CLI_OPTS");
+            String scanCliOptsEnvironment = intEnvironmentVariables.getValue("SCAN_CLI_OPTS");
             if (StringUtils.isNotBlank(scanCliOptsEnvironment)) {
                 scanCliOptsToUse = scanCliOptsEnvironment;
             }
         }
-        final boolean commandDryRun = blackDuckUrl == null || dryRun;
+        boolean commandDryRun = blackDuckUrl == null || dryRun;
         String commandScheme = null;
         String commandHost = null;
         int commandPort = 0;
         if (!commandDryRun) {
-            commandScheme = blackDuckUrl.getProtocol();
-            commandHost = blackDuckUrl.getHost();
-            if (blackDuckUrl.getPort() > 0) {
-                commandPort = blackDuckUrl.getPort();
-            } else if (blackDuckUrl.getDefaultPort() > 0) {
-                commandPort = blackDuckUrl.getDefaultPort();
+            commandScheme = blackDuckUrl.url().getProtocol();
```

Comment:
Rather than call the .url() method over and over we could do that before the if block and use that value in these places instead